pkg/snapshot/snapshot.go

```diff
@@ -0,0 +1,389 @@
+
+			if err := e.Encode(&apiDef, bw); err != nil {
+				return err
+			}
+
+			if _, err := bw.WriteString("\n---\n\n"); err != nil {
+				return err
+			}
+		}
+
+		if policiesFile != "" {
+			policyFile, err := os.Create(policiesFile)
+			if err != nil {
+				return err
+			}
+			defer policyFile.Close()
+
+			pw := bufio.NewWriter(policyFile)
+			if err := writePolicies(policiesList, pw, e); err != nil {
+				return err
+			}
+		}
+
+		return bw.Flush()
+	}
+
+	// Output file will contain ApiDefinition based on specified category.
+	category = strings.TrimSpace(category)
+	if !strings.HasPrefix(category, "#") {
+		category = fmt.Sprintf("#%s", category)
+	}
+
+	fmt.Printf("Looking for ApiDefinitions in %s category.\n", category)
+
+	i := 0
+
+	for _, v := range apiDefSpecList.Apis {
+		if contains := strings.Contains(v.Name, category); !contains {
+			continue
+		}
+
+		// Parse Config Data of the ApiDefinition created on Dashboard.
+		name, ns := parseConfigData(v, fmt.Sprintf("%s-%d", DefaultName, i))
+
+		// create an ApiDefinition object.
+		apiDef := createApiDef(name, ns)
+		apiDef.Spec.APIDefinitionSpec = *v
+
+		storeMetadata(apiDef.Spec.APIID, apiDef.ObjectMeta.Name, apiDef.ObjectMeta.Namespace)
+		i++
+
+		if err := e.Encode(&apiDef, bw); err != nil {
+			return err
+		}
+
+		if _, err := bw.WriteString("\n---\n\n"); err != nil {
+			return err
+		}
+	}
+
+	if policiesFile != "" {
+		policyFile, err := os.Create(policiesFile)
+		if err != nil {
+			return err
+		}
+		defer policyFile.Close()
+
+		pw := bufio.NewWriter(policyFile)
+		if err := writePolicies(policiesList, pw, e); err != nil {
+			return err
+		}
+	}
+
+	return bw.Flush()
+}
+
+func createApiDef(metaName, metaNs string) tykv1alpha1.ApiDefinition {
+	return tykv1alpha1.ApiDefinition{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ApiDefinition",
+			APIVersion: "tyk.tyk.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      metaName,
+			Namespace: metaNs,
+		},
+		Spec: tykv1alpha1.APIDefinitionSpec{},
+	}
+}
+
+func storeMetadata(key, name, namespace string) {
+	names[key] = name
+	nameSpaces[key] = namespace
+}
+
+func getMetadata(key string) (name, namespace string) {
+	return names[key], nameSpaces[key]
+}
+
+// fetchPolicies lists all SecurityPolicy objects form dashboard.
+func fetchPolicies(e *environmet.Env) ([]tykv1alpha1.SecurityPolicySpec, error) {
+	url := client.JoinURL(e.URL, "/api/portal/policies?p=-2")
+	method := http.MethodGet
+
+	hc := &http.Client{}
+
+	req, err := http.NewRequest(method, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Authorization", e.Auth)
+
+	res, err := hc.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	policiesResponse := &dashboard.PoliciesResponse{}
+
+	err = jsonEncoding.NewDecoder(res.Body).Decode(&policiesResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	return policiesResponse.Policies, nil
+}
```

Comment:
This logic belongs in the Dashboard Client All method

Reply:
@asoorm, `All` method of Policies returns paged results and It seems like the current client does not support fetching results page by page - or I am missing something.

As far as I understand from the policies API, I need to give `?p=-2` query parameter to fetch all policies from the dashboard. That's why I added a separate function with a specific query parameter. 

Should we update our client in a separate PR to fetch all policies without paging?

Reply:
I would suggest that the current client is a bug - and we should edit the current client to return ALL policies using `?p=-2`

Reply:
Created an issue on GH: #503 
I will raise a PR asap.